Docs Audit #2 (#3801)
dfinity/portal: docs/developer-docs/smart-contracts/maintain/logs.mdx

- Queries, only if called in replicated mode.

> A canister's controller is able to retrieve the canister's logs, even when executions trap, using [`dfx canister logs`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-logs):

Reviewer: Could you clarify that the controller needs to run this command?

```
dfx canister logs <canister-name>
```

:::info
Non-replicated calls are not currently supported for logging.
:::

## Log visibility

By default, a canister's logs are only visible to the controllers of the canister. A canister's logs can be made public by updating the canister's settings with [`dfx canister update-settings`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-update-settings) command:

```bash
dfx canister update-settings <canister-name> --log-visibility public
```

### Log viewer allow lists

Canister log allow lists enable principals that are not controllers of the canister to view the logs without the logs being public. Log allow lists are supported in `dfx` versions `0.24.2` and newer.

> To add a principal to the canister's log allow list, update the canister's settings with the flags:

Reviewer 2: ```suggestion
To configure the canister's log allow list, update the canister's settings with the flags:
```

> - `dfx canister update-settings <canister-name> --set-log-viewer PRINCIPAL_ID`: Set a single principal as a log viewer.

Author: ```suggestion
- `dfx canister update-settings <canister-name> --set-log-viewer <principal-id>`: Set a single principal as a log viewer.
```

> - `dfx canister update-settings <canister-name> --add-log-viewer PRINCIPAL_ID`: Add a principal to a list of log viewers.

Author: ```suggestion
- `dfx canister update-settings <canister-name> --add-log-viewer <principal-id>`: Add a principal to a list of log viewers.
```

> - `dfx canister update-settings <canister-name> --remove-log-viewer PRINCIPAL_ID`: Remove a principal from the log viewer list.

Author: ```suggestion
- `dfx canister update-settings <canister-name> --remove-log-viewer <principal-id>`: Remove a principal from the log viewer list.
```

You can also configure a principal as a log viewer when you create a canister:

```

> dfx canister create <canister-name> --log-viewer PRINCIPAL_ID

Author: ```suggestion
dfx canister create <canister-name> --log-viewer <principal-id>
```

```

> Or, you can set the `canisters[].initialization_values.log_visibility.allowed_viewers` configuration setting in your project's `dfx.json` file.

Reviewer 2: when will this setting have an effect? would be good to clarify this

> Or, you can set the `canisters[].initialization_values.log_visibility.allowed_viewers` configuration setting in your project's `dfx.json` file.

Author: ```suggestion
Or, you can set the `canisters[].initialization_values.log_visibility.allowed_viewers` configuration setting in your project's `dfx.json` file to set the canister log allowed viewers upon canister creation.
```
